### PR TITLE
Add ability to support multiple feed services

### PR DESCRIPTION
**Pull request by jtapia**
Quick Info
---
Migrations? :+1:

What does this change?
----
- Add support for `Google Feed`, `Google Merchant Center` and `Facebook Feed`
- Add admin `product_feed` and `product_catalog` controller and test case
- Add spree `product_feed` and `product_catalog` model and test case
- Add spree `product_feed` and `product_catalog` table[migration]
- Add `product_feed` and `product_catalog` views
- Fix `product_feed` and `product_catalog` styles
- Add logic to handle the type of the response(xml or csv)

How do you manually test this?
- Go to `admin` section
- Click on `navbar` icon or go to `/admin/product_feeds`
----

Screenshots
----
http://recordit.co/y8jD1VjX8i

### CODE REVIEW

**Comment by fastjames** [comment]
From watching the screencast, it looks like the store owner gets to choose one format and that determines what `/products.rss` looks like. Is that correct?

**Comment by jtapia** [comment]
@fastjames I found out that facebook supports the xml schema, so I changed the logic to not set a format, I just updated the screencast and also now is pointing to `gemini_master `, thanks

**Comment by fastjames** [comment]
[ch1416]

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ nbproject
 spec/dummy
 pkg
 Gemfile.lock
+coverage/
```
Comment (joshcampbell) [inline .gitignore]: What tool creates this?

Comment (fastjames) [reply .gitignore]: I think that's `simplecov`.

File: app/controllers/spree/products_controller_decorator.rb [hunk 2]
```diff
@@ -1,18 +1,44 @@
-Spree::ProductsController.prepend(Module.new do
-  class << self
-    def prepended(klass)
-      klass.respond_to :rss, :xml, only: :index
+Spree::ProductsController.class_eval do
```
[the remark this answers is not in the document]
Comment (jtapia) [reply app/controllers/spree/products_controller_decorator.rb]: Not really, I consider that we don't need to define a `respond_to` method, instead we can use a `before_action`, what do you think?

**Comment by fastjames** [comment]
Merged.

**Comment by aneekam** [comment]
https://app.clubhouse.io/enginecommerce/story/1138/product-feeds-should-include-variants-rather-than-products

**Pull request closed without merging**